libraries/AP_Scripting/generator/description/bindings.desc

```diff
@@ -152,3 +152,14 @@ singleton AP_SerialLED method send void
 include SRV_Channel/SRV_Channel.h
 singleton SRV_Channels alias SRV_Channels
 singleton SRV_Channels method find_channel boolean SRV_Channel::Aux_servo_function_t'enum SRV_Channel::k_none SRV_Channel::k_nr_aux_servo_functions-1 uint8_t'Null
+
+include AP_Mission/AP_Mission.h
+singleton AP_Mission alias mission
+singleton AP_Mission enum MISSION_STOPPED MISSION_RUNNING MISSION_COMPLETE
+singleton AP_Mission method state uint8_t
+singleton AP_Mission method get_current_nav_index uint16_t
+singleton AP_Mission method set_current_cmd boolean uint16_t 0 UINT16_MAX
```

Comment:
This has a race condition, as the vehicle code could be trying to change the command as well, which could lead to corruption.

All the getters are technically not thread safe, but I'm not to bothered by them as they are reading a single value back, which should be reasonably atomic.

Reply:
Any tips on how to fix? Can I just add a with semaphore to set_current_cmd?

Reply:
If you add a semaphore (ideally a recursive one) to the library (and then have the libraries helpers that mutate state also take it, that would require no changes in the vehicle code, or in the scripting side.

Your other option (which I really don't like, but is the prior art we have) still requires you a semaphore in the AP_Mission library, but the vehicle is responsible for taking the semaphore manually before making any calls into the library. The advantage of this is the vehicle can hold the semaphore for a longer timer if it knows it's going to be doing some back to back calls, the downside (and this is why I don't like it) is that all calling code needs to be responsible for calling it. If you go with this approach you can simply add this to your description file (and regenerate):
```
singleton AP_Mission semaphore
```
then all the mission calls will automatically call `get_semaphore()` on the library and take/release the semaphore appropriately.

I'd push for the first option here as the mission stuff is sorta deeply strewn around the vehicle code and MAVLink code. Even the AHRS example from before actually is only taken in AP_AHRS.h and in libraries, as trying to get the protection everywhere in the vehicle code is a path to insanity :)

Reply:
added with_semaphore to set_current_command, I had overlooked that mission already had its own semaphore